rust-src/concordium_base/src/transactions.rs

```diff
@@ -263,7 +263,70 @@ pub struct TransactionHeader {
     pub expiry: TransactionTime,
 }
 
-#[derive(Debug, Clone, SerdeSerialize, SerdeDeserialize, Into, AsRef)]
+#[derive(Debug, Clone, SerdeSerialize, SerdeDeserialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+/// Header V1 of an account transaction. Same as TransactionHeader but contains
+/// an additional optional sponsor address.
+pub struct TransactionHeaderV1 {
+    /// Sender account of the transaction.
+    pub sender: AccountAddress,
+    /// Sequence number of the transaction.
+    pub nonce: Nonce,
+    /// Maximum amount of energy the transaction can take to execute.
+    pub energy_amount: Energy,
+    /// Size of the transaction payload. This is used to deserialize the
+    /// payload.
+    pub payload_size: PayloadSize,
+    /// Latest time the transaction can be included in a block.
+    pub expiry: TransactionTime,
+    /// Optional transaction sponsor.
+    pub sponsor: Option<AccountAddress>,
+}
+
+impl Serial for TransactionHeaderV1 {
+    fn serial<B: Buffer>(&self, out: &mut B) {
+        let bitmap: u16 = if self.sponsor.is_some() {
+            0b0000000000000001
+        } else {
+            0b0000000000000000
+        };
+        let header_v0 = TransactionHeader {
+            sender: self.sender,
+            nonce: self.nonce,
+            energy_amount: self.energy_amount,
+            payload_size: self.payload_size,
+            expiry: self.expiry,
+        };
+        bitmap.serial(out);
+        header_v0.serial(out);
+        if let Some(sponsor) = self.sponsor {
+            sponsor.serial(out);
+        }
+    }
+}
+
+impl Deserial for TransactionHeaderV1 {
+    fn deserial<R: ReadBytesExt>(source: &mut R) -> ParseResult<Self> {
+        let bitmap: u16 = source.get()?;
```

Comment:
We also need to check if any unknown bits are set in the bitmap, and return an error if that's the case